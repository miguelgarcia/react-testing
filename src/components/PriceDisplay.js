export function PriceDisplay({ crypto, price }) {
  return <div>{`Current price of ${crypto} is ${price}`}</div>;
}
