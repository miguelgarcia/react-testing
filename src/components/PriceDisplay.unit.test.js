import { render, screen } from '@testing-library/react';
import { PriceDisplay } from './PriceDisplay';

describe('PriceDisplay', () => {
  test('PriceDisplay shows info from the props', () => {
    render(<PriceDisplay crypto='ETH' price='20.5' />);
    const linkElement = screen.getByText(/Current price of ETH is 20.5/i);
    expect(linkElement).toBeInTheDocument();
  });

  test('PriceDisplay regression test', () => {
    const tree = render(<PriceDisplay crypto='ETH' price='20.5' />).asFragment();
    expect(tree).toMatchSnapshot();
  });
});
