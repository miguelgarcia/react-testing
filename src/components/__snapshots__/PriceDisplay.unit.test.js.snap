// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceDisplay PriceDisplay regression test 1`] = `
<DocumentFragment>
  <div>
    Current price of ETH is 20.5
  </div>
</DocumentFragment>
`;
